crates/node/src/scheduler/mod.rs

```diff
@@ -245,12 +247,25 @@ impl Scheduler {
                 }
             }
 
-            let mut task = match self.pick_task().await {
-                Some(t) => {
+            if let Err(err) = watchdog_sender
+                .send(HealthCheckSignal::SchedulerLoopOk)
+                .await
+            {
+                tracing::error!("Watchdog channel send return an error:{err}");
+            }
+
+            let (mut task, mempool_size) = match self.pick_task().await {
+                (Some(t), size) => {
                     tracing::debug!("task {}/{} scheduled for running", t.id, t.tx);
-                    t
+                    (t, size)
                 }
-                None => {
+                (None, size) => {
+                    if let Err(err) = watchdog_sender
+                        .send(HealthCheckSignal::SchedulerMempoolLen(0))
```

Comment:
Double-check: Should the parameter here be `0` or `size` ?

Reply:
Yes, good catch, it's size.